content/blog/2017/2017-02-14-declarative-notifications.adoc

```diff
@@ -0,0 +1,436 @@
+and `post` blocks.  In one of those `post&#123; }`  blocks, we included a placeholder for
+sending notifications.
+
+In this blog post, we'll pull in the notification method we created for
+link:/blog/2016/07/18/pipline-notifications/[Sending Notifications in Pipeline].
+Then we'll move that method into a shared library,
+letting us call it from other projects and keeping our `Jenkinsfile` concise and understandable.
+
+== Setup
+
+The setup for this post is a almost the same as the
+link:/blog/2017/02/10/declarative-html-publisher/[previous post].
+I've used a new branch
+link:https://github.com/bitwiseman/hermann[my same fork] of the
+link:https://github.com/reiseburo/hermann[hermann project]:
+link:https://github.com/bitwiseman/hermann/tree/blog/declarative/notifications[`blog/add-declarative/notifications`].
+I've set up a Multibranch Pipeline and pointed it at my repository
+the same as did it previous post.
+Also the same as before, I've set this Pipeline's Git configuration to
+automatically "Clean after checkout".
+
+In addition, I still have my notification targets setup from the
+"link:/blog/2016/07/18/pipline-notifications/[Sending Notifications in Pipeline]" blog post.
+Take a look there for help setting up the
+plugin:slack[Slack],
+plugin:hipchat[HipChat],
+and plugin:email-ext[Email-ext]
+plugins.
+
+
+Here's the final form of the Declarative Pipeline from the previous post:
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+  agent {
+    // Use docker container
+    docker {
+      image 'ruby:2.3'
+    }
+  }
+  options {
+    // Only keep the 10 most recent builds
+    buildDiscarder(logRotator(numToKeepStr:'10'))
+  }
+  stages {
+    stage ('Install') {
+      steps {
+        // install required gems
+        sh 'bundle install'
+      }
+    }
+    stage ('Build') {
+      steps {
+        // build
+        sh 'bundle exec rake build'
+      }
+
+      post {
+        success {
+          // Archive the built artifacts
+          archive includes: 'pkg/*.gem'
+        }
+      }
+    }
+    stage ('Test') {
+      steps {
+        // run tests with coverage
+        sh 'bundle exec rake spec'
+      }
+
+      post {
+        success {
+          // publish html
+          publishHTML target: [
+              allowMissing: false,
+              alwaysLinkToLastBuild: false,
+              keepAll: true,
+              reportDir: 'coverage',
+              reportFiles: 'index.html',
+              reportName: 'RCov Report'
+            ]
+        }
+      }
+    }
+  }
+  post {
+    always {
+      echo "Send notifications for result: ${currentBuild.result}"
+    }
+  }
+}
+// Scripted //
+----
+
+
+== Preparing for Notifications
+
+The Pipeline above has a final `always` block that prints a message to the status log,
+but doesn't print anything at the start of the run.
+The final form of the Pipeline in the original blog post called the same method twice,
```

Comment:
"The final form" makes me think of Pokemon. it also seems a little repetitive since the previous statement used "final" too. Why not just say "The (Scripted) Pipeline created in the original blog post"